Documentation/reference-guides/xfrm/index.rst

```diff
@@ -0,0 +1,301 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _xfrm_guide:
+
+********************
+XFRM Reference Guide
+********************
+
+.. note:: This documentation section is targeted at developers and users who
+          want to understand the Linux XFRM subsystem. While reading this
+          reference guide may help broaden your understanding of Cilium, it is
+          not a requirement to use Cilium. Please refer to the
+          :ref:`getting_started` guide and :ref:`ebpf_datapath` for a higher
+          level introduction.
+
+Overview
+========
+
+IPsec encryption in the Linux kernel relies on `XFRM`_. XFRM is an IP framework intended for packet transformations, from encryption to compression. It is configured via a set of *policy* and *state* objects, which for IPsec, correspond to Security Policies and Security Associations.
+
+.. _XFRM: https://man7.org/linux/man-pages/man8/ip-xfrm.8.html
+
+XFRM Policies and States
+------------------------
+
+At a high-level, XFRM policies define what to accept and reject, whereas states define how to perform the encryption and decryption. Policies can match on the direction (``out``, ``in``, or ``fwd``), the source and destination IP addresses with CIDRs, and the packet mark. As an example, the following policy matches egressing packets with any source IP address, 10.56.1.X destination IP addresses, and ``0xcb93eXX`` packet marks. Policies default to allowing traffic as done here.
+
+.. code-block:: text
+
+    src 0.0.0.0/0 dst 10.56.1.0/24 
+        dir out priority 0 
+        mark 0xcb93e00/0xffffff00 
+        [...]
+
+States are relatively similar, except that they are agnostic to the direction and can only match on exact IP addresses (or 0.0.0.0 to match all). The following state will apply to packets with IP addresses 10.56.0.17 -> 10.56.1.238, the same packet marks as above. For ingressing, encrypted packets, the SPI will also be used (discussed below). 
```

Comment:
You may want to express what these source and destination IP addresses are. Which I guess leads a bit into how an SA is defined. Each source and dst represent one side of an IPSec secured tunnel.